### DISCUSSION TITLE
why is `next.config.js` treated as a client component?

### DISCUSSION THREAD

**Discussion by stefanprobst** [vercel/next.js#65605] [Help]
### Summary

i am trying to import a module which has `import "server-only";` into `next.config.js` and get the following error:

```
 ⨯ Failed to load next.config.js, see more info here https://nextjs.org/docs/messages/next-config-error

> Build error occurred
Error: This module cannot be imported from a Client Component module. It should only be used from a Server Component.
```

i had assumed that `next.config.js` would not be shipped to the client - is this not the case?

### Example

https://github.com/stefanprobst/issue-next-config-server-import

**Comment by nayounsang** [comment]
I am curious about this issue to.
[**next.config.js is a regular Node.js module, not a JSON file. It gets used by the Next.js server and build phases, and it's not included in the browser build.**](https://nextjs.org/docs/app/api-reference/next-config-js)
Could "dynamic import inside conditional statement" be your solution?

**Reply by stefanprobst** [reply]
hi, unfortunately, dynamic import did not work.

i also checked whether `next.config.js` is actually read client-side by adding the following:

```ts
if (typeof document !== "undefined") {
  throw new Error("Reading next.config.js on the client.");
}
```

and everything still seemed to work. (see [this branch](https://github.com/stefanprobst/issue-next-config-server-import/blob/next-config-client-throw/next.config.js))

**Comment by stefanprobst** [comment]
opened https://github.com/vercel/next.js/issues/65652 about this
